docs/guides/style/terraform-style-guide.md

```diff
@@ -131,7 +131,7 @@ There is no standard on ordering the sections, but as a rule of thumb the follow
 - Provider blocks, if needed.
 - The main component of the module (e.g., the `aws_ecs_service` resource for the ECS service module).
 - All other sections.
-- Any `data` blocks (at the bottom).
+- `locals` and `data` blocks just before where they're used.
```

Comment:
I think it depends on both cases. I tend to keep both blocks on the top and consolidate as much into one `local {}` as possible, if they're producing values that are used in more than one resource/module call. For an AMI lookup or a very specific string calculation that requires a `local` to be used, those are cases I think putting them directly above what is going to use them next helps the reader.

Reply:
One thought: we used to separate out all data sources to a separate `dependencies.tf` file so that users can scan what resources we expect to exist for the module and are looked up (this pattern is employed to some extent in `terraform-aws-eks` repo). Similarly, one of the contractors pointed out that it might be better for `locals` to be separated out into a separate `locals.tf` file to make it easier to scan, search, and find.

I personally think that might be a better practice than having everything in one giant `main.tf` file at the top. Thoughts on that approach as an alternative?

Reply:
I'm okay with either way. I find it easier to scan and search if they're in the same file, but I could see myself creating a plugin or shortcut that opens all *.tf files automagically into separate window-splits. I tend not to remember things when flipping between files that aren't visible at the same time.

For long term, separate files would be cleaner.